tests/fdb/test_fdb.py

```diff
@@ -150,25 +151,46 @@ def setup_fdb(ptfadapter, vlan_table, router_mac, pkt_type):
     time.sleep(FDB_POPULATE_SLEEP_TIMEOUT)
     # Flush dataplane
     ptfadapter.dataplane.flush()
+
     return fdb
 
 
-@pytest.fixture
+def get_fdb_dynamic_mac_count(duthost):
+    res = duthost.command('show mac')
+    logger.info('"show mac" output on DUT:\n{}'.format(pprint.pformat(res['stdout_lines'])))
+    total_mac_count = 0
+    for l in res['stdout_lines']:
+        if "dynamic" in l.lower():
+            total_mac_count += 1
+    return total_mac_count
+
+
 def fdb_cleanup(duthosts, rand_one_dut_hostname):
     """ cleanup FDB before and after test run """
     duthost = duthosts[rand_one_dut_hostname]
-    try:
-        duthost.command('sonic-clear fdb all')
-        yield
-    finally:
-        # in any case clear fdb after test
+    total_dyn_mac_count = get_fdb_dynamic_mac_count(duthost)
+    if total_dyn_mac_count == 0:
+        return
+    else:
+        done = False
         duthost.command('sonic-clear fdb all')
+        while not done:
+            total_dyn_mac_count = get_fdb_dynamic_mac_count(duthost)
+            if total_dyn_mac_count != 0:
+                time.sleep(FDB_CLEAN_UP_SLEEP_TIMEOUT)
+            else:
+                return
```

Comment:
I think this section could be refactored to use `wait_until`. This does require us to set an upper-bound on how long we'll poll, but I think that's probably a good thing to have in the (hopefully unlikely :smile:) case we hit some bug and the dynamic MAC count never hits 0.

Reply:
@daall Agreed. I have made the changes to use wait_until() and pytest_assert().